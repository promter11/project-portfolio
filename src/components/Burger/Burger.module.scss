@import "../../styles/_variables";
@import "../../styles/_mixins";

.burger {
  position: absolute;
  top: 3rem;
  left: calc(100% + 3rem);
  width: 2rem;
  cursor: pointer;

  @media (min-width: 993px) {
    display: none;
  }

  .item {
    @include width-height(100%, 0.2rem);
    background-color: $white-color;

    &.itemEnter,
    &.itemExitActive,
    &.itemExitDone {
      transition: transform 300ms;

      &:first-child,
      &:last-child {
        transform: translate(0);
      }

      &:nth-child(2) {
        opacity: 1;
      }
    }

    &.itemEnterActive,
    &.itemEnterDone,
    &.itemExit {
      transition: transform 300ms;

      &:first-child {
        transform: translate(0, 0.7rem) rotate(-45deg);
      }

      &:nth-child(2) {
        opacity: 0;
      }

      &:last-child {
        transform: translate(0, -0.7rem) rotate(45deg);
      }
    }

    &:not(:last-child) {
      margin-bottom: 0.5rem;
    }
  }
}
