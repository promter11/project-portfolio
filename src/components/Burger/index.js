import React from "react";
import { CSSTransition } from "react-transition-group";

import styles from "./Burger.module.scss";

const Burger = ({ active, onSetActive }) => {
  return (
    <ul className={styles.burger} onClick={() => onSetActive(!active)}>
      {Array.from({ length: 3 }).map((item, index) => {
        return (
          <CSSTransition
            key={index}
            in={active}
            timeout={300}
            classNames={{
              enter: styles.itemEnter,
              enterActive: styles.itemEnterActive,
              enterDone: styles.itemEnterDone,
              exit: styles.itemExitActive,
              exitActive: styles.itemExitActive,
              exitDone: styles.itemExitDone,
            }}
          >
            <li className={styles.item} />
          </CSSTransition>
        );
      })}
    </ul>
  );
};

export default Burger;
