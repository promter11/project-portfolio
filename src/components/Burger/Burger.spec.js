import React from "react";
import { shallow } from "enzyme";

import Burger from "./index";

const props = {
  status: false,
  onToggleStatus: jest.fn(),
};

describe("Burger component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Burger {...props} />);
  });

  it("renders correctly with props", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("simulates onClick event on onToggleStatus function", () => {
    wrapper.find("ul").simulate("click");

    expect(props.onToggleStatus).toHaveBeenCalled();
  });
});
