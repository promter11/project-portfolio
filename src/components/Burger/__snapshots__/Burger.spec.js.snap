// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger component renders correctly with props 1`] = `
<ul
  className="burger"
  onClick={[Function]}
>
  <CSSTransition
    classNames={
      Object {
        "enter": "itemEnter",
        "enterActive": "itemEnterActive",
        "enterDone": "itemEnterDone",
        "exit": "itemExitActive",
        "exitActive": "itemExitActive",
        "exitDone": "itemExitDone",
      }
    }
    in={false}
    key="0"
    timeout={300}
  >
    <li
      className="item"
    />
  </CSSTransition>
  <CSSTransition
    classNames={
      Object {
        "enter": "itemEnter",
        "enterActive": "itemEnterActive",
        "enterDone": "itemEnterDone",
        "exit": "itemExitActive",
        "exitActive": "itemExitActive",
        "exitDone": "itemExitDone",
      }
    }
    in={false}
    key="1"
    timeout={300}
  >
    <li
      className="item"
    />
  </CSSTransition>
  <CSSTransition
    classNames={
      Object {
        "enter": "itemEnter",
        "enterActive": "itemEnterActive",
        "enterDone": "itemEnterDone",
        "exit": "itemExitActive",
        "exitActive": "itemExitActive",
        "exitDone": "itemExitDone",
      }
    }
    in={false}
    key="2"
    timeout={300}
  >
    <li
      className="item"
    />
  </CSSTransition>
</ul>
`;
